cpp/test/prims/batched/batched_csr.cu

```diff
@@ -0,0 +1,208 @@
+
+    // Copy the data to the device
+    updateDevice(AbM.raw_data(), A.data(), A.size(), stream);
+    updateDevice(BxbM.raw_data(), Bx.data(), Bx.size(), stream);
+
+    // Create sparse matrix A from the dense A and the mask
+    BatchedCSR<T> AbS = BatchedCSR<T>::from_dense(AbM, mask, cusolverSpHandle);
+
+    // Create matrix that will hold the results
+    res_bM = new LinAlg::Batched::BatchedMatrix<T>(m_r, n_r, params.batch_size,
+                                                   handle, allocator, stream);
+
+    // Compute the tested results
+    switch (params.operation) {
+      case SpMV_op:
+        b_spmv((T)1.0, AbS, BxbM, (T)0.0, *res_bM);
+        break;
+      case SpMM_op:
+        b_spmm((T)1.0, AbS, BxbM, (T)0.0, *res_bM);
+        break;
+    }
+
+    // Compute the expected results
+    res_h.resize(params.batch_size * m_r * n_r);
+    switch (params.operation) {
+      case SpMV_op:
+        for (int bid = 0; bid < params.batch_size; bid++) {
+          LinAlg::Naive::naiveMatMul(
+            res_h.data() + bid * m_r, A.data() + bid * params.m * params.n,
+            Bx.data() + bid * params.p, params.m, params.n, 1);
+        }
+        break;
+      case SpMM_op:
+        for (int bid = 0; bid < params.batch_size; bid++) {
+          LinAlg::Naive::naiveMatMul(res_h.data() + bid * m_r * n_r,
+                                     A.data() + bid * params.m * params.n,
+                                     Bx.data() + bid * params.p * params.q,
+                                     params.m, params.n, params.q);
+        }
+        break;
+    }
+
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+  }
+
+  void TearDown() override {
+    delete res_bM;
+    CUBLAS_CHECK(cublasDestroy(handle));
+    CUDA_CHECK(cudaStreamDestroy(stream));
+    CUSOLVER_CHECK(cusolverSpDestroy(cusolverSpHandle));
+  }
+
+ protected:
+  BatchedCSRInputs<T> params;
+  LinAlg::Batched::BatchedMatrix<T> *res_bM;
+  std::vector<T> res_h;
+  cublasHandle_t handle;
+  cusolverSpHandle_t cusolverSpHandle;
+  cudaStream_t stream;
+};
+
+// Test parameters (op, batch_size, m, n, nnz, p, q, tolerance)
+const std::vector<BatchedCSRInputs<double>> inputsd = {
+  {SpMV_op, 1, 90, 150, 440, 150, 1, 1e-6},
+  {SpMV_op, 5, 13, 12, 75, 12, 1, 1e-6},
+  {SpMV_op, 15, 8, 4, 6, 4, 1, 1e-6},
+  {SpMV_op, 33, 7, 7, 23, 7, 1, 1e-6},
+  {SpMM_op, 1, 20, 15, 55, 15, 30, 1e-6},
+  {SpMM_op, 9, 10, 9, 31, 9, 11, 1e-6},
+  {SpMM_op, 20, 7, 12, 11, 12, 13, 1e-6}};
+
+// Test parameters (op, batch_size, m, n, nnz, p, q, tolerance)
+const std::vector<BatchedCSRInputs<float>> inputsf = {
+  {SpMV_op, 1, 90, 150, 440, 150, 1, 1e-2},
+  {SpMV_op, 5, 13, 12, 75, 12, 1, 1e-2},
+  {SpMV_op, 15, 8, 4, 6, 4, 1, 1e-2},
+  {SpMV_op, 33, 7, 7, 23, 7, 1, 1e-2},
+  {SpMM_op, 1, 20, 15, 55, 15, 30, 1e-2},
+  {SpMM_op, 9, 10, 9, 31, 9, 11, 1e-2},
+  {SpMM_op, 20, 7, 12, 11, 12, 13, 1e-2}};
+
+using BatchedCSRTestD = BatchedCSRTest<double>;
```

Comment:
```suggestion
using CSRTestD = CSRTest<double>;
```

Reply:
Apparently Google Benchmark doesn't want the fixtures to have the same name even if in in different namespaces (I had a bug like that with `BatchedMatrix`). There is no fixture named `CSRTestD` in the non-batched test yet, but I think it's less trouble to have distinct name for this test now and avoid trouble later.